modules/data/src/test/java/io/fluo/webindex/data/util/LinkUtilTest.java

```diff
@@ -28,6 +28,12 @@ public void testBasic() throws Exception {
     Assert.assertTrue(LinkUtil.isValid("http://1.2.3.4/"));
     Assert.assertTrue(LinkUtil.isValid("http://a.b.c.d.com/1/2/3/4/5"));
     Assert.assertTrue(LinkUtil.isValid("http://a.b.com:281/1/2"));
+    Assert.assertTrue(LinkUtil.isValid("http://A.B.Com:281/a/b"));
+    Assert.assertTrue(LinkUtil.isValid("http://A.b.Com:281/A/b"));
+    Assert.assertTrue(LinkUtil.isValid("http://a.B.Com?A/b/C"));
+    Assert.assertTrue(LinkUtil.isValid("http://A.Be.COM"));
+    Assert.assertTrue(LinkUtil.isValid("http://a.com:/test"));
+    Assert.assertTrue(LinkUtil.isValid("http://z.com:"));
```

Comment:
In addition to calling isValid.  Should test also check expected output of cleanUrl?  

```java
Assert.assertEquals("http://z.com", cleanUrl("http://z.com:"));
```